### ISSUE TITLE
Unify the functions for pre-processing 2D images and 3D volumes

### ISSUE THREAD

**Issue by junchaoxia** [ComputationalCryoEM/ASPIRE-Python#122]
The pre-processing functions from the original ASPIRE-python under src/aspire/aspire/preprocessor.py have different implementations with the internal functions of Image and Volume classes. This make comparison and tests more difficult to performed.  For example, the rotation matrices to generate the 2D images from a 3D map have the x and y axes reversed, 
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/master/src/aspire/volume/__init__.py#L87 
We need to unify all functions related to these pre-processing operations.

**Comment by junchaoxia** [comment]
In PR #190, the preprocess steps in the pipeline of new framework have included phase flip, downsample,  normalizing background, whiten, and inverting contrast. There might be some other improvements in related open issues and we will solve them later. The issue of projection from 3D map to 2D was also mentioned in a separate open issue #196. Closing now.

**Comment by janden** [comment]
We still have two versions of these functions, right? To complete the unification, we need to remove the old ones and make sure that the new ones are called instead.

**Comment by garrettwrong** [comment]
@chris-langfield , I assigned to you because this might have already been addressed (I know you did some reconciliation for downsampling methods).

Can you try to see what about this ticket is still relevant? If there are still actionable tasks lets jot them down.  The projection convention can be disregarded because there is another issue for that already.  Thanks

**Comment by chris-langfield** [comment]
The two functions in `image/preprocess.py` which are *used* are `crop_pad` and `downsample`. 

These both have new and better tested analogues:

`aspire.utils.coor_trans.crop_pad_2d`, `aspire.utils.coor_trans.crop_pad_3d`, and `Image.downsample()` and `Volume.downsample()`. These are all tested independently of the old versions in

`test_coor_trans.py` and `test_downsample`. 

2D downsampling was discussed in #140, #566, and #608. There you can find discussions of matching the MATLAB implementation and a bugfix. The new code was introduced in #567 and #606.

3D downsampling was discussed in #617. #679 and #689 refactored the Volume downsampling code. The behavior *exactly* matches the old `preprocess.py` version. There is an open issue #710 but this was not introduced, but rather uncovered, by this work.

Based on this I'm comfortable deleting the implementations in `preprocess.py`. This only advantage they currently have is (apparently) supporting stacks, but our new methods can be updated to do this if we want. 

`fuzzy_mask()` is the only other function in this file. It is tested (with a hardcoded test) in `test_preprocess.py` but not used elsewhere. I think this could easily go into either a `utils` file or into `Image`, if we even want to keep it. 

TLDR:

- Delete old `crop_pad` and `downsample` implementations.
- Delete or move `fuzzy_mask` since we don't use it.
- Delete the entire `preprocess.py` file after doing the above

**Comment by chris-langfield** [comment]
#773 will close
